algosdk/transaction.py

```diff
@@ -2503,7 +2503,7 @@ def __eq__(self, other):
         )
 
 
-class LogicSig:
+class _LogicSig:
```

Comment:
@ahangsu + reviewers - The following remark might be off-base.  I feel like I don't have a good handle (yet) on `LogicSig` vs `LogicSigAccount` use cases.

`_LogicSig` is exposed in several public methods/fields (e.g. `LogicSigAccount.lsig`, `LogicSigTransaction.__init__`).  Consequently:
* I don't feel confident marking the class private is sufficient.  Though I'm open to being told I'm off-base.
* I _think_ we'd need to remove all public references to `_LogicSig`.  Since I don't feel I understand the usage patterns, I'm unsure how palatable the change feels.

Reply:
I agree that `LogicSig` should not be hidden/removed from the SDK. I think deprecating the whole class was too coarse grain. We need `LogicSig` to represent the logic signature structure for a transaction, i.e. `LogicSigTransaction.lsig`.

However, we don't need need to make it possible to directly construct `LogicSig`s now that we have `LogicSigAccount`. Perhaps functions like `LogicSig.sign` and `LogicSig.append_to_multisig` can be deprecated, hidden, or removed.

Reply:
I still feel fuzzy and may benefit from a verbal - Here's my understanding:
* `LogicSig` is still safe to use for _some_ use cases (e.g. escrow).
* `LogicSig` is _not_ fully compatible with all delegated logicsig use cases (e.g. rekeying like in https://github.com/algorand/go-algorand-sdk/pull/230).  Here, `LogicSigAccount` fills the gap and additionally supports all `LogicSig` use cases.

If the above statements are accurate, then I think:
* I still don't know enough to recommend alternatives.  I imagine there's alternate object modelings that better represent desired behaviors + prevent undesired behaviors (like using `LogicSig` in an incompatible way).  
* I'd like someone to confirm the worst case behavior here.  As I understand it, an unintended combination of `LogicSig` + signing address results in a runtime exception.  If so, it feels _OK_.
* Assuming others see the topic as _large_, I propose:
  * Remove deprecation on `LogicSig` and keep it for v2.0.0. 
  * Extend `LogicSig` docstring to qualify when to use `LogicSig` vs `LogicSigAccount`.  Possibly with a similar note in `LogicSig`.
    * If we take the proposal, I'd appreciate help in writing the commentary.  I still don't think I fully grok the distinctions.

Reply:
I think the suggestion sounds reasonable:
* We want to discourage public usage of `LogicSig`, but the deprecation comment might be a bit misleading (since it is still internally used by other public-facing methods). Adding commentary sounds prudent.
* I'm also okay with keeping everything as-is (state before this PR) as a path of least resistance. The behavior was a bit opaque to me, but it doesn't seem incorrect.

Reply:
That sounds reasonable to me as well